azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub_mqtt.py

```diff
@@ -115,22 +119,22 @@ def _handle_pipeline_event(self, event):
         if isinstance(event, pipeline_events_mqtt.IncomingMessage):
             topic = event.topic
 
-            if mqtt_topic.is_c2d_topic(topic):
+            if mqtt_topic_iothub.is_c2d_topic(topic):
                 message = Message(event.payload)
-                mqtt_topic.extract_properties_from_topic(topic, message)
+                mqtt_topic_iothub.extract_properties_from_topic(topic, message)
                 self.handle_pipeline_event(pipeline_events_iothub.C2DMessageEvent(message))
 
-            elif mqtt_topic.is_input_topic(topic):
+            elif mqtt_topic_iothub.is_input_topic(topic):
                 message = Message(event.payload)
-                mqtt_topic.extract_properties_from_topic(topic, message)
-                input_name = mqtt_topic.get_input_name_from_topic(topic)
+                mqtt_topic_iothub.extract_properties_from_topic(topic, message)
```

Comment:
Not related to this PR, but I really hate the `extract_properties_from_topic` function for reasons outlined in comments in `mqtt_topic_iothub.py` and it should be refactored at some point.

Reply:
Yes!  I want to see that file refactored, rewritten to use URI-parsing functions, and put directly under test.  

---
In reply to: [287151503](https://github.com/Azure/azure-iot-sdk-python-preview/pull/95#discussion_r287151503) [](ancestors = 287151503)